sdk/tables/azure-tables/review/tables.api.review.md

```diff
@@ -0,0 +1,695 @@
+}
+
+// @public
+export type TableQueryEntitiesWithPartitionAndRowKeyResponse = TableQueryEntitiesWithPartitionAndRowKeyHeaders & {
+  [propertyName: string]: any;
+} & {
+  _response: coreHttp.HttpResponse & {
+    bodyAsText: string;
+    parsedBody: {
+      [propertyName: string]: any;
+    };
+    parsedHeaders: TableQueryEntitiesWithPartitionAndRowKeyHeaders;
+  };
+};
+
+// @public
+export interface TableQueryHeaders {
+  clientRequestId?: string;
+  date?: Date;
+  requestId?: string;
+  version?: string;
+  xMsContinuationNextTableName?: string;
+}
+
+// @public
+export interface TableQueryOptionalParams extends coreHttp.OperationOptions {
+  nextTableName?: string;
+  queryOptions?: GeneratedQueryOptions;
+  requestId?: string;
+}
+
+// @public
+export interface TableQueryResponse {
+  odataMetadata?: string;
+  value?: TableResponseProperties[];
+}
+
+// @public
+export type TableResponse = TableResponseProperties & {
+  odataMetadata?: string;
+};
+
+// @public
+export interface TableResponseProperties {
+  odataEditLink?: string;
+  odataId?: string;
+  odataType?: string;
+  tableName?: string;
+}
+
+// @public
+export class TableServiceClient {
+  constructor(url: string, options?: TableServiceClientOptions);
+  createEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    options?: CreateEntityOptions
+  ): Promise<CreateEntityResponse>;
+  createTable(tableName: string, options?: CreateTableOptions): Promise<CreateTableResponse>;
+  deleteEntity(
+    tableName: string,
+    partitionKey: string,
+    rowKey: string,
+    ifMatch: string,
+    options?: DeleteEntityOptions
+  ): Promise<DeleteEntityResponse>;
+  deleteTable(tableName: string, options?: DeleteTableOptions): Promise<DeleteTableResponse>;
+  static fromConnectionString(
+    connectionString: string,
+    options?: TableServiceClientOptions
+  ): TableServiceClient;
+  getAccessPolicy(
+    tableName: string,
+    options?: GetAccessPolicyOptions
+  ): Promise<GetAccessPolicyResponse>;
+  getEntity<T extends object>(
+    tableName: string,
+    partitionKey: string,
+    rowKey: string,
+    options?: GetEntityOptions
+  ): Promise<GetEntityResponse<T>>;
+  getProperties(options?: GetPropertiesOptions): Promise<GetPropertiesResponse>;
+  getStatistics(options?: GetStatisticsOptions): Promise<GetStatisticsResponse>;
+  listEntities<T extends object>(
+    tableName: string,
+    query?: QueryOptions,
+    options?: ListEntitiesOptions
+  ): Promise<ListEntitiesResponse<T>>;
+  listTables(query?: QueryOptions, options?: ListTablesOptions): Promise<ListTablesResponse>;
+  setAccessPolicy(
+    tableName: string,
+    acl?: SignedIdentifier[],
+    options?: SetAccessPolicyOptions
+  ): Promise<SetAccessPolicyResponse>;
+  setProperties(
+    properties: ServiceProperties,
+    options?: SetPropertiesOptions
+  ): Promise<SetPropertiesResponse>;
+  updateEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    mode?: UpdateMode,
+    eTag?: string,
```

Comment:
If we are already passing the `entity`, do we need to also pass the `eTag`? Or can we just extract it from the entity we were passed?

Reply:
Do we want to default eTag to `"*"` or do we make it required and don't assume `"*"` on undefined?

`"*"` will force unconditional update

Reply:
From the arch board review, we'd like to have etag as optional and default to "*"